File: 2023/UICPC DIV1/UICPC Round #06/solutions & data/forestforthetrees/submissions/accepted/trees-db.py
#!/bin/python3

from fractions import gcd

[x,y] = [int(x) for x in input().split()]
[x1,y1,x2,y2] = [int(x) for x in input().split()]

g = gcd(x,y)

X, Y = x//g, y//g

if x1 <= X and X <= x2 and y1 <= Y and Y <= y2:
    k = min(x2//X, y2//Y) + 1
    X *= k
    Y *= k

if x <= X:
    print("Yes")
else:
    print("No")
    print(X,Y)

File: 2023/UICPC DIV1/UICPC Round #06/solutions & data/forestforthetrees/submissions/accepted/trees-db.cc
#include <bits/stdc++.h>
using namespace std;

int main(){
  long long x,y; cin >> x >> y;

  long long ax,ay,bx,by;
  cin >> ax >> ay >> bx >> by;

  long long g = __gcd(x,y);
  long long X = x/g, Y = y/g;

  if(ax <= X && X <= bx && ay <= Y && Y <= by){
    long long k = min(bx/X, by/Y) + 1;
    X *= k;
    Y *= k;
  }
  
  if(X < x){
    assert(Y < y);
    cout << "No" << endl;
    cout << X << " " << Y << endl;
  } else {
    cout << "Yes" << endl;
  }
}
